src/Build/BackEnd/Shared/TargetResult.cs

```diff
@@ -136,6 +136,21 @@ public TargetResultCode ResultCode
             }
         }
 
+        public string TargetResultCodeToString()
+        {
+            switch (ResultCode)
+            {
+                case TargetResultCode.Failure:
+                    return "Failure";
```

Comment:
```suggestion
                    return nameof(TargetResultCode.Failure);
```

This is more resilient to future refactorings